src/canister/individual_user_template/src/api/hot_or_not_bet/reenqueue_timers_for_pending_bet_outcomes.rs

```diff
@@ -18,6 +25,88 @@ pub fn reenqueue_timers_for_pending_bet_outcomes() {
     });
 }
 
+#[update]
+// #[update(guard = "is_caller_global_admin_or_controller")]
+async fn once_reenqueue_timers_for_pending_bet_outcomes() -> Result<Vec<(u64, u8)>, String> {
+    let current_time = system_time::get_current_system_time_from_ic();
+
+    let post_w_slot = CANISTER_DATA.with(|canister_data_ref_cell| {
+        let canister_data = canister_data_ref_cell.borrow_mut();
+
+        let posts_with_slots =
+            once_get_posts_that_have_pending_outcomes(&canister_data, &current_time);
+
+        // once_reenqueue_timers_for_these_posts(&mut canister_data, posts_with_slots, &current_time);
+        once_reenqueue_timers_for_these_posts(posts_with_slots.clone());
+        posts_with_slots
+    });
+
+    Ok(post_w_slot)
+}
+
+fn once_reenqueue_timers_for_these_posts(post_slot_ids: Vec<(u64, u8)>) {
+    for (post_id, slot_id) in post_slot_ids {
+        let slot_number = slot_id;
+        // let jitter = rand::thread_rng().gen_range(1..60);
+
+        ic_cdk_timers::set_timer(
+            // random jitter
+            Duration::from_secs(300),
+            move || {
+                CANISTER_DATA.with(|canister_data_ref_cell| {
+                    tabulate_hot_or_not_outcome_for_post_slot(
+                        &mut canister_data_ref_cell.borrow_mut(),
+                        post_id,
+                        slot_number,
+                    );
+                });
+            },
+        );
+    }
+}
+
+fn once_get_posts_that_have_pending_outcomes(
+    canister_data: &CanisterData,
+    current_time: &SystemTime,
+) -> Vec<(u64, u8)> {
+    let room_details_map = &canister_data.room_details_map;
+    canister_data
+        .all_created_posts
+        .iter()
+        .rev()
+        .flat_map(|(post_id, post)| {
+            let created_outside_the_last_48_hours = current_time
```

Comment:
edge case : are we not doing for posts within 48 hrs whose slots have passed ?

Reply:
I think that is taken care by [the post_upgrade](https://github.com/go-bazzinga/hot-or-not-backend-canister/blob/13fe85ce32e89cab3a60411d13091893edcbbfd1/src/canister/individual_user_template/src/api/hot_or_not_bet/reenqueue_timers_for_pending_bet_outcomes.rs#L146-L162).

Reply:
post_upgrade will only launch timers from that time. so in between slots wont get processed.

Reply:
again an edge case, so take a call if that needs to be handled

Reply:
an easy way would be to run the platofrom orch reenque function again after few days

Reply:
we will have to run this once only, to cover this edge case. so this should be fine, right?